modules/light-clients/07-tendermint/types/misbehaviour_handle.go

```diff
@@ -32,6 +34,32 @@ func (cs ClientState) CheckMisbehaviourAndUpdateState(
 
 	// The status of the client is checked in 02-client
 
+	// if heights are equal check that this is valid misbehaviour of a fork
+	// otherwise if heights are unequal check that this is valid misbehavior of BFT time violation
+	if tmMisbehaviour.Header1.GetHeight().EQ(tmMisbehaviour.Header2.GetHeight()) {
+		blockID1, err := tmtypes.BlockIDFromProto(&tmMisbehaviour.Header1.SignedHeader.Commit.BlockID)
+		if err != nil {
+			return nil, sdkerrors.Wrap(err, "invalid block ID from header 1 in misbehaviour")
+		}
+		blockID2, err := tmtypes.BlockIDFromProto(&tmMisbehaviour.Header2.SignedHeader.Commit.BlockID)
+		if err != nil {
+			return nil, sdkerrors.Wrap(err, "invalid block ID from header 2 in misbehaviour")
+		}
+
+		// Ensure that Commit Hashes are different
+		if bytes.Equal(blockID1.Hash, blockID2.Hash) {
+			return nil, sdkerrors.Wrap(clienttypes.ErrInvalidMisbehaviour, "headers block hashes are equal")
+		}
+	} else {
+		// Header1 is at greater height than Header2, therefore Header1 time must be less than or equal to
```

Comment:
just to confirm, do we check this somewhere?

Reply:
We check it on `ValidateBasic` of Misbehaviour